src/functionaltests/java/com/ericsson/gerrit/plugins/eiffel/linking/LinkingSteps.java

```diff
@@ -0,0 +1,449 @@
+        event.put("id", id);
+        JSONArray events = new JSONArray();
+        events.put(event);
+        JSONObject body = new JSONObject();
+        body.put("events", events);
+        return body;
+    }
+
+    private Supplier<PatchSetAttribute> getPatSetAttribute() {
+        return getPatSetAttribute(NOT_USED);
+    }
+
+    private Supplier<PatchSetAttribute> getPatSetAttribute(String commitSha) {
+        PatchSetAttribute patchSetAttribute = mock(PatchSetAttribute.class);
+        patchSetAttribute.author = new AccountAttribute();
+        boolean shouldHaveRevision = !commitSha.equals(NOT_USED);
+        if (shouldHaveRevision) {
+            patchSetAttribute.revision = commitSha;
+        }
+
+        @SuppressWarnings("unchecked")
+        Supplier<PatchSetAttribute> patchSetAttributeSupplier = mock(Supplier.class);
+        when(patchSetAttributeSupplier.get()).thenReturn(patchSetAttribute);
+        return patchSetAttributeSupplier;
+    }
+
+    private Supplier<ChangeAttribute> getChangeAttribute(String projectName, String branch) {
+        ChangeAttribute changeAttribute = new ChangeAttribute();
+        changeAttribute.project = projectName;
+        changeAttribute.branch = branch;
+        @SuppressWarnings("unchecked")
+        Supplier<ChangeAttribute> changeAttributeSupplier = mock(Supplier.class);
+        when(changeAttributeSupplier.get()).thenReturn(changeAttribute);
+        return changeAttributeSupplier;
+    }
+
+    private void callListenersOnEvent(PatchSetEvent patchSetEvent) {
+        for (AbstractEventListener abstractEventListener : listeners) {
+            abstractEventListener.onEvent(patchSetEvent);
+        }
+        try {
+            waitFor(SECOND_10);
+        } catch (InterruptedException e) {
+            throw new RuntimeException("Could not wait for events to be sent");
+        }
+    }
+
+    private void waitFor(int maxWaitTimeMillis) throws InterruptedException {
+        ThreadPoolExecutor executor = Whitebox.<ThreadPoolExecutor>getInternalState(
+                AbstractEventListener.class, "executor");
+
+        /*
+         * There is a small possibility that we can reach this place before the executor has started
+         * a thread
+         */
+        boolean finished = false;
+        long stopTime = System.currentTimeMillis() + maxWaitTimeMillis;
+        while (!finished && stopTime > System.currentTimeMillis()) {
+            if (executor.getActiveCount() > 0) {
+                Thread.sleep(SECOND_1);
+            } else {
+                finished = true;
+            }
+        }
+    }
+
+    private static Map<String, String> getEventMap() {
+        Map<String, String> myMap = new HashMap<String, String>();
+        myMap.put("SCC", "EiffelSourceChangeCreatedEvent");
+        myMap.put("SCS", "EiffelSourceChangeSubmittedEvent");
+        return myMap;
+    }
+
+    private class ModuleDependencis extends AbstractModule {
```

Comment:
This class should maybe be outside this test class.

Reply:
The class is only used here